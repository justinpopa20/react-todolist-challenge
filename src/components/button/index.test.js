import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Button from './'

describe('Button', () => {
  test('should render', () => {
    const button = render(<Button type="add">Test</Button>)

    expect(button.baseElement).toMatchSnapshot()
  })

  test('should fire onclick', () => {
    const clickSpy = jest.fn()
    const wrapper = render(<Button type="add" onClick={clickSpy} data-testid="btn">Test</Button>)
    const button = wrapper.getByTestId("btn")

    fireEvent.click(button)

    expect(clickSpy).toHaveBeenCalled()
  })
})