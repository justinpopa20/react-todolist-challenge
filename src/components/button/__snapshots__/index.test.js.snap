// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<body>
  <div>
    <button
      class="button add"
    >
      Test
    </button>
  </div>
</body>
`;
